draft-ietf-taps-interface.md

```diff
@@ -737,23 +737,60 @@ early binding when required, for example with some Network Address Translator
 
 ### Using Multicast Endpoints
 
-Specifying a multicast group address on a Local Endpoint will indicate to the Transport
-Services system that the resulting connection will be used to receive multicast messages. The
-Remote Endpoint can be used to filter incoming multicast from specific senders. Such
-a Preconnection will only support calling Listen(), not Initiate(). Calling Listen()
-will cause the Transport Services system to register for receiving multicast, such
-as issuing an IGMP join {{?RFC3376}} or using MLD for IPV6 {{?RFC4604}}. Any Connections
-that are accepted from this Listener are receive-only.
-
-Similarly, specifying a multicast group address on the Remote Endpoint will indicate that the
-resulting connection will be used to send multicast messages, and that the Preconnection will
-support Initiate() but not Listen(). Any Connections created this way are send-only.
-
-A Rendezvous() call on Preconnections containing group addresses results in an
-EstablishmentError as described in {{rendezvous}}.
+To use multicast, a Preconnection is first created with the Local/Remote Endpoint
+specifying the ASM/SSM multicast group and destination port number.
+
+Calling Initiate() on that Preconnection creates a Connection that can be
+used to send messages to the multicast group. The Connection object that is
+created will support Send() but not Receive(). Any Connections created this
+way are send-only, and do not join the multicast group. The resulting
+Connection will have a Local Endpoint indicating the local interface to
+which the connection is bound and a Remote Endpoint indicating the
+multicast group.
+
+```
+RemoteSpecifier.WithMulticastGroupIPv4(GroupAddress)
+RemoteSpecifier.WithMulticastGroupIPv6(GroupAddress)
+RemoteSpecifier.WithPort(PortNumber)
+```
+
+Calling Listen() on a Preconnection with a multicast group specified on the Remote
+Endpoint will join the multicast group to receive messages. This Listener
+will create one Connection for each Remote Endpoint sending to the group,
+with the Local Endpoint set to the group address. The set of Connection
+objects created forms a Connection Group.
+
```

Comment:
I think this makes sense, but I wonder if the Local Endpoint must somehow specify both the group address and the local address/port to which the connection is bound? 

Or, to put this another way, when calling `Listen()` is it a requirement to bind to a specific local interface? If not, does the system pick a default interface from which to join the group, or does it join the group from all available interfaces? If the former, how can the application know what interface was used? If the latter, how can it tell on what interface each packet was received?

Reply:
I discussed this with @GrumpyOldTroll in Philly … and we decided to leave that unspecified. We assume the implementation will throw an error if it has insufficient information. In the examples, we alway give an Interface, but a source address might also be sufficient.

Reply:
```suggestion
The receiving interface can be restricted by passing it as part of the LocalSpecifier or queried through the MessagContext on the messages received (see {{msg-ctx}} for further details).

```